### PR TITLE
Adding back -in-element and -with-dynamic-vars

### PR DESCRIPTION
**Pull request by chadhietala**

### CODE REVIEW

File: packages/@glimmer/test-helpers/lib/environment.ts [hunk 1]
```diff
@@ -1086,34 +1086,6 @@ function populateBlocks(blocks: BlockMacros, inlines: InlineMacros): { blocks: B
     builder.evaluate(sexp[5]);
   });
 
-  blocks.add('-with-dynamic-vars', (sexp, builder) => {
```
Comment (rwjblue) [inline packages/@glimmer/test-helpers/lib/environment.ts]: Is this removal related to adding `in-element`?

Comment (chadhietala) [inline packages/@glimmer/test-helpers/lib/environment.ts]: Both of these were removed from the environment, but for some reason implemented just in the test environment.

Comment (chadhietala) [inline packages/@glimmer/test-helpers/lib/environment.ts]: This puts them back as first class things that an environment gets by default.

Comment (rwjblue) [inline packages/@glimmer/test-helpers/lib/environment.ts]: 👍
